src/render/md.jl

```diff
@@ -102,5 +149,69 @@ function writemd(io::IO, m::Meta{:source})
 end
 
 function headermd(io::IO, doc::Metadata, config::Config)
-    print_help(io, config.mdstyle_header, doc.modname)
+    println_mdstyle(io, config.mdstyle_header, doc.modname)
+end
+
+## Docs-specific rendering
+function writemd(io::IO, docs::Docs{:md})
+    println(io, docs.data)
+end
+
+### API-Index ----------------------------------------------------------------------------
+
+type ApiIndex
+    sourcepaths :: Vector{String}
+    modulenames :: Vector{String}
+    entriesmd   :: Vector{EntriesMD}
+
+    ApiIndex() = new(Vector{String}([]), Vector{String}([]), Vector{EntriesMD}([]))
+end
+
+const API_INDEX = ApiIndex()
+
+function savegenindex(file::String, mime::MIME"text/md", c::Config)
+    # Write the API-Index file.
+    genindexfiledir = dirname(abspath(file))
+    isfile(file) || mkpath(genindexfiledir)
+    open(file, "w") do f
+        info("writing API-Index to $(file)")
+        println_mdstyle(f, c.mdstyle_header, "API-INDEX\n")
+        println(f)
+
+        for i in 1:length(API_INDEX.sourcepaths)
+            relsourcepath = relpath(API_INDEX.sourcepaths[i], genindexfiledir)
+            println_mdstyle(f, c.mdstyle_genindex_mod,
+                "$(c.md_genindex_modprefix)$(API_INDEX.modulenames[i])\n")
+
+            process_api_index(f, "Exported", API_INDEX.entriesmd[i].exported, relsourcepath, c);
+            process_api_index(f, "Internal", API_INDEX.entriesmd[i].internal, relsourcepath, c)
+        end
+    end
+    clear_api_index(API_INDEX)
+end
+
+function clear_api_index(api_index::ApiIndex)
```

Comment:
`clear_api_index` -> `clear_api_index!` to make clear that it mutates an argument if we keep the global approach for this.